org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -382,7 +387,7 @@ private <T> Stream<CompletableFuture<T>> executeOnServers(
 	 * forever...
 	 */
 	private <T> void completeEmptyOrWithException(final CompletableFuture<Optional<T>> completableFuture, final Throwable t) {
-		if (t != null) {
+		if (t != null && !isRequestCancelledException(t)) {
```

Comment:
@ahmedneilhussain , I think this will do for the case computeFirst, which is enough for this PR, but it would be nice if it would work the same way for collectAll and computeAll, but I am not quite sure on how to code it. Do you have good ideas for that?